course-content/22_cleaning_it_up.qmd

```diff
@@ -0,0 +1,410 @@
+
+Pandas has functions specifically designed to cast datatypes from one to another, as was covered in the previous section with the `astype()` function.  
+
+To cast a variable as a datetime datatype, we can use the pandas `to_datetime()` function and pass in the pandas series you want to modify:
+
+```{python}
+m_health['timestamp'] = pd.to_datetime(m_health['timestamp'])
+m_health.head()
+m_health.info()
+```
+
+Et voila - timestamp is now of the datatime64 datatype! Were we wanting to leverage timestamp's date time qualities, we could now do so more easily. 
+
+
+### Recoding values
+
+Value recoding is a data cleaning task is to put these values into more manageable categories. This is a common data recoding task that most data analyst type folks will be familiar with. Pandas alone has many functions that solve recoding issues in different ways. We will just highlight a few in this tutorial.  
+
+With this dataset we've called there are a few columns that we would like to make changes to so that it will be easier to analyze the data in the way we want. The `value_counts()` method shows the range of responses and their frequency in the dataframe. 
+
+```{python}
+m_health['gender'].value_counts()
+```
+
+With that knowledge, we can utilize the `replace()` function and pass in custom lists with all of the responses that should be replaced by a single specfied value. 
+
+```{python}
+m_health['gender'].replace(['Male ', 'male', 'M', 'm', 'Male', 'Cis Male'],
+                            'Male', inplace=True)
+m_health['gender'].replace(['Female ', 'female', 'F'],
+                            'Female', inplace=True)
+m_health['gender'].value_counts()
+```
+
+Another task is to put categorical objects into a form that is more amenable to computational tasks. Let's take a look at a categorial variable that consists of responses of "Yes" and "No".
+
+```{python}
+m_health['self_employed'].value_counts()
+```
+
+If we want to leverage this data purely as a segment to split other metrics with, we would probably leave as is. But perhaps we would like to treat it as a dummy variable also so we can do more sophisticated math with it. 
+
+For our self-employed variable, we begin by implementing `replace()` to transform the existing content to numbers as shown below:
+
+
+```{python}
+m_health['self_employed'].replace(['Yes'], '1', inplace=True)
+m_health['self_employed'].replace(['No'], '0', inplace=True)
+m_health['self_employed'].value_counts()
+```
+
+The pandas `astype()` function is able to interpret the 1s and 0s as integers and transform the series into an `int` datatype.
```

Comment:
```suggestion
The pandas `astype()` function is able to interpret the 1s and 0s as integers and so we can transform the series into an `int` datatype.
```

Clarifying that this has to be done manually - pandas doesn't make it int on its own.